pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -275,7 +275,7 @@ cdef convert_to_timedelta64(object ts, str unit):
             ts = cast_from_unit(ts, unit)
             ts = np.timedelta64(ts, "ns")
     elif isinstance(ts, str):
-        if len(ts) > 0 and ts[0] == "P":
+        if len(ts) > 0 and (ts[0] == "P" or ts[:2] == "-P"):
```

Comment:
you need another length check here (e.g. if  len(ts) ==1 this would raise)

Reply:
updated